File: code/python/1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashMap = {num:index for index,num in enumerate(nums)}
        res = []
        for i,num in enumerate(nums):
            if target-num in hashMap and i!=hashMap[target-num]:
                return [i,hashMap[target-num]]
        return res

File: code/java/1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> map = new HashMap<>();
        for(int i=0;i<nums.length;i++){
            if(!map.containsKey(nums[i]))
                map.put(nums[i],i);
        }
        for(int i=0;i<nums.length;i++){
            if(map.containsKey(target-nums[i])&&i!=map.get(target-nums[i]))
                return new int[]{i,map.get(target-nums[i])};
        }
        return new int[]{0};
    }
}